lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/participants/diagnostics/VersionValidator.java

```diff
@@ -22,24 +23,35 @@
 import org.eclipse.lemminx.extensions.maven.utils.MavenParseUtils;
 import org.eclipse.lsp4j.Diagnostic;
 import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4j.jsonrpc.CancelChecker;
 
 class VersionValidator {
 
 	private static final Logger LOGGER = Logger.getLogger(VersionValidator.class.getName());
 
-	public static Optional<List<Diagnostic>> validateVersion(DiagnosticRequest diagnosticRequest) {
+	CancelChecker cancelChecker;
+	
+	public VersionValidator(CancelChecker cancelChecker) {
+		this.cancelChecker = cancelChecker;
+	}
+	
+	public Optional<List<Diagnostic>> validateVersion(DiagnosticRequest diagnosticRequest) throws CancellationException {
+		cancelChecker.checkCanceled();
 		DOMNode node = diagnosticRequest.getNode();
 		Dependency model = MavenParseUtils.parseArtifact(node);
 		Artifact artifact = null;
 		try {
+			cancelChecker.checkCanceled();
 			artifact = new DefaultArtifact(model.getGroupId(), model.getArtifactId(), model.getVersion(), model.getScope(), model.getType(), model.getClassifier(), new DefaultArtifactHandler(model.getType()));
 			if (!artifact.isSelectedVersionKnown()) {
+				cancelChecker.checkCanceled();
 				return Optional.of(Collections
 						.singletonList(diagnosticRequest.createDiagnostic("Version Error", DiagnosticSeverity.Error)));
 			}
 		} catch (Exception e) {
```

Comment:
You will catch CancelException here, you need to catch it and rethrow it and not log it.

Reply:
It looks like this validator currently isn't used, but you're right.